doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -425,16 +425,98 @@ Displaying information about notations
+
+   For example, this output from `Print Grammar tactic` shows the first 3 levels for
+   `tactic_expr`, designated as "5", "4" and "3".  Level 3 is right-associative,
+   which applies to the productions within it, such as the `try` construct::
+
+     Entry tactic:tactic_expr is
+     [ "5" RIGHTA
+       [ tactic:binder_tactic ]
+     | "4" LEFTA
+       [ SELF; ";"; tactic:binder_tactic
+       | SELF; ";"; SELF
+       | SELF; ";"; tactic_then_locality; tactic_then_gen; "]" ]
      | "3" RIGHTA
        [ IDENT "try"; SELF
+       :
+
+   The interpretation of `SELF` depends on its position in the production and the
+   associativity of the level:
+
+   - At the beginning of a production, `SELF` means the next level.  In the
+     fragment shown above, the next level for `try` is "2".  (This is defined by the order
+     of appearance in the grammar or output; the levels could just as well be
+     named "foo" and "bar".)
+   - In the middle of a production, `SELF` means the top level ("5" in the fragment)
+   - At the end of a production, `SELF` means the next level within
+     `LEFTA` levels and the current level within `RIGHTA` levels.
+
+   `NEXT` always means the next level. `nonterminal LEVEL "…"` is a reference to the specified level
+   for `nonterminal`.
+
+   `Associativity <http://camlp5.github.io/doc/htmlc/grammars.html#b:Associativity>`_
+   explains `SELF` and `NEXT` in somewhat more detail.
+
+   The output for `Print Grammar constr` includes :cmd:`Notation` definitions,
+   which are dynamically added to the grammar at run time.
+   For example, in the definition for `operconstr`, the production on the second line shown
+   here is defined by a :cmd:`Reserved Notation` command in `Notations.v`::
+
+     | "50" LEFTA
+       [ SELF; "||"; NEXT
+
+   Similarly, `Print Grammar tactic` includes :cmd:`Tactic Notation`\s, such as :tacn:`dintuition`.
+
+   The file
+   `doc/tools/docgram/fullGrammar <http://github.com/coq/coq/blob/master/doc/tools/docgram/fullGrammar>`_
+   in the source tree extracts the full grammar for
+   |Coq| (not including notations and tactic notations defined in `*.v` files nor some optionally-loaded plugins)
+   in a single file with minor changes to handle nonterminals using multiple levels (described in
+   `doc/tools/docgram/README.md <http://github.com/coq/coq/blob/master/doc/tools/docgram/README.md>`_).
+   This is complete and much easier to read than the grammar source files.
+   `doc/tools/docgram/orderedGrammar <http://github.com/coq/coq/blob/master/doc/tools/docgram/orderedGrammar>`_
+   has the edited grammar that's used in the documentation.
 
-   Note that the productions printed by this command are represented in the form used by
-   |Coq|'s parser (coqpp), which differs from how productions are shown in the documentation.
+   Developer documentation for parsing is in
+   `doc/parsing.md <http://github.com/coq/coq/blob/master/doc/parsing.md>`_.
```

Comment:
```suggestion
   `dev/doc/parsing.md <http://github.com/coq/coq/blob/master/dev/doc/parsing.md>`_.
```

Reply:
Updated, ready to go, thanks.  The TODOs are intentional.